android/src/main/java/com/rt2zz/reactnativecontacts/ContactsManager.java

```diff
@@ -565,7 +565,7 @@ public void updateContact(ReadableMap contact, Callback callback) {
         ops.add(op.build());
 
         op = ContentProviderOperation.newUpdate(ContactsContract.Data.CONTENT_URI)
-                .withSelection(ContactsContract.Data.CONTACT_ID + "=?", new String[]{String.valueOf(recordID)})
+                .withSelection(ContactsContract.Data.CONTACT_ID + "=? AND " + ContactsContract.Data.MIMETYPE + " = ?", new String[]{String.valueOf(recordID), StructuredName.CONTENT_ITEM_TYPE})
```

Comment:
I dont quite understand the code. Will you lightly explain how this fixes things?

Reply:
Also, let's upstream this too

Reply:
Yup, working on description now